src/sortStudents.ts

```diff
@@ -1,16 +1,78 @@
 
 export interface Student {
-  // describe Student interface
+  name: string,
+  surname: string,
+  age: number,
+  married: boolean,
+  grades: number[],
 }
 
 export enum SortType {
-  // describe SortType enum
+  Name = 'name',
+  Surname = 'surname',
+  Age = 'age',
+  Married = 'married',
+  AverageGrade = 'grades',
 }
 
-// create SortOrder type
-export type SortOrder;
+export type SortOrder = 'asc' | 'desc';
 
+export function sortStudents(
+  students: Student[],
+  sortBy: SortType,
+  order: SortOrder,
+): Student[] {
+  const copyOfStudents = [...students];
 
-export function sortStudents(students, sortBy, order) {
-  // write your function
+  function averageValue(person: Student): number {
+    return person.grades.reduce((sum, item) => sum + item)
+    / person.grades.length;
+  }
+
+  switch (sortBy) {
+    case SortType.Name:
+      return (order === 'asc')
+        ? copyOfStudents.sort((firstStudent, secondStudent) => firstStudent
+          .name.localeCompare(secondStudent.name))
+
+        : copyOfStudents.sort((firstStudent, secondStudent) => secondStudent
+          .name.localeCompare(firstStudent.name));
+
+    case SortType.Surname:
+      return (order === 'asc')
+        ? copyOfStudents.sort((firstStudent, secondStudent) => firstStudent
+          .surname.localeCompare(secondStudent.surname))
+
+        : copyOfStudents.sort((firstStudent, secondStudent) => secondStudent
+          .surname.localeCompare(firstStudent.surname));
+
+    case SortType.Age:
+      return (order === 'asc')
+        ? copyOfStudents.sort((firstStudent, secondStudent) => firstStudent
+          .age - secondStudent.age)
+
+        : copyOfStudents.sort((firstStudent, secondStudent) => secondStudent
+          .age - firstStudent.age);
+
+    case SortType.Married:
+      return (order === 'asc')
+        ? copyOfStudents.sort((firstStudent, secondStudent) => +firstStudent
+          .married - +secondStudent.married)
```

Comment:
Use `Number()` in such cases, it's more obvious.
```suggestion
        ? copyOfStudents.sort((firstStudent, secondStudent) => Number(firstStudent
          .married) - Number(Student.married))
```